File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const map = {}
    let res = [-1, -1];

    nums.forEach((ele, index) => {
        if (map[target - ele] != undefined) {
            res = [map[target - ele], index];
            return;
        }
        map[ele] = index;
    })

    return res;
};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res = {-1, -1};
        int n = nums.size();
        unordered_map<int, int> mp;

        for(int i = 0; i < n; i++) {
            if(mp.find(target - nums[i]) != mp.end()) {
                res[0] = mp[target - nums[i]];
                res[1] = i;
                break;
            }

            mp[nums[i]] = i;
        }

        return res;
    }
};